lambdas/call-dcmaw-async-cri/src/test/java/uk/gov/di/ipv/core/calldcmawasynccri/CallDcmawAsyncCriHandlerTest.java

```diff
@@ -0,0 +1,167 @@
+        assertEquals(ErrorResponse.MISSING_IPV_SESSION_ID.getCode(), lambdaResult.get("code"));
+        assertEquals(
+                ErrorResponse.MISSING_IPV_SESSION_ID.getMessage(), lambdaResult.get("message"));
+    }
+
+    @Test
+    void handleRequestShouldReturnJourneyErrorResponseIfDcmawCriServiceThrows() throws Exception {
+        when(mockIpvSessionService.getIpvSession("a-session-id")).thenReturn(mockIpvSessionItem);
+        when(mockClientOAuthSessionDetailsService.getClientOAuthSession(any()))
+                .thenReturn(new ClientOAuthSessionItem());
+        when(mockDcmawAsyncCriService.startDcmawAsyncSession(any(), any(), any()))
+                .thenThrow(new RuntimeException("Oh dear"));
+
+        Map<String, Object> lambdaResult =
+                callDcmawAsyncCriHandler.handleRequest(input, mockContext);
+
+        assertEquals("/journey/error", lambdaResult.get("journey"));
+        assertEquals(HttpStatus.SC_INTERNAL_SERVER_ERROR, lambdaResult.get("statusCode"));
+        assertEquals(
+                ErrorResponse.ERROR_CALLING_DCMAW_ASYNC_CRI.getCode(), lambdaResult.get("code"));
+        assertEquals(
+                ErrorResponse.ERROR_CALLING_DCMAW_ASYNC_CRI.getMessage(),
+                lambdaResult.get("message"));
+    }
+
+    @Test
+    void handleRequestShouldReturnJourneyErrorResponseIfCiStoringServiceThrows() throws Exception {
+        List<Exception> exceptionsToThrow =
+                List.of(new SqsException("Oops"), new RuntimeException("Oops"));
+
+        for (Exception e : exceptionsToThrow) {
+            when(mockIpvSessionService.getIpvSession("a-session-id"))
+                    .thenReturn(mockIpvSessionItem);
+            when(mockClientOAuthSessionDetailsService.getClientOAuthSession(any()))
+                    .thenReturn(clientOAuthSessionItem);
+            when(mockVerifiableCredentialResponse.getUserId()).thenReturn(TEST_USER_ID);
+            when(mockVerifiableCredentialResponse.getCredentialStatus())
+                    .thenReturn(VerifiableCredentialStatus.PENDING);
+            when(mockDcmawAsyncCriService.startDcmawAsyncSession(any(), any(), any()))
+                    .thenReturn(mockVerifiableCredentialResponse);
+            doThrow(e).when(mockCriStoringService).storeCriResponse(any(), any(), any(), any());
+
+            Map<String, Object> lambdaResult =
+                    callDcmawAsyncCriHandler.handleRequest(input, mockContext);
+
+            assertEquals("/journey/error", lambdaResult.get("journey"));
+            assertEquals(HttpStatus.SC_INTERNAL_SERVER_ERROR, lambdaResult.get("statusCode"));
+            assertEquals(
+                    ErrorResponse.ERROR_CALLING_DCMAW_ASYNC_CRI.getCode(),
+                    lambdaResult.get("code"));
+            assertEquals(
+                    ErrorResponse.ERROR_CALLING_DCMAW_ASYNC_CRI.getMessage(),
+                    lambdaResult.get("message"));
+        }
+    }
+}
```

Comment:
The code does check that the returned user ID is the expected one, so a test for that would be good.